src/Compilers/CSharp/Portable/Symbols/SourceEmbeddedAttributeSymbol.cs

```diff
@@ -0,0 +1,177 @@
+
+        internal override bool HasDeclarativeSecurity => false;
+
+        internal override bool IsInterface => false;
+
+        internal override NamedTypeSymbol BaseTypeNoUseSiteDiagnostics => _baseType;
+
+        internal override ObsoleteAttributeData ObsoleteAttributeData => null;
+
+        public override ImmutableArray<Symbol> GetMembers() => _members;
+
+        public override ImmutableArray<Symbol> GetMembers(string name) => _members.Where(member => member.Name == name).ToImmutableArray();
+
+        public override ImmutableArray<CustomModifier> GetTypeArgumentCustomModifiers(int ordinal) => ImmutableArray<CustomModifier>.Empty;
+
+        public override ImmutableArray<NamedTypeSymbol> GetTypeMembers() => ImmutableArray<NamedTypeSymbol>.Empty;
+
+        public override ImmutableArray<NamedTypeSymbol> GetTypeMembers(string name) => ImmutableArray<NamedTypeSymbol>.Empty;
+
+        public override ImmutableArray<NamedTypeSymbol> GetTypeMembers(string name, int arity) => ImmutableArray<NamedTypeSymbol>.Empty;
+
+        internal override ImmutableArray<string> GetAppliedConditionalSymbols() => ImmutableArray<string>.Empty;
+
+        internal override AttributeUsageInfo GetAttributeUsageInfo() => AttributeUsageInfo.Default;
+
+        internal override NamedTypeSymbol GetDeclaredBaseType(ConsList<Symbol> basesBeingResolved) => _baseType;
+
+        internal override ImmutableArray<NamedTypeSymbol> GetDeclaredInterfaces(ConsList<Symbol> basesBeingResolved) => ImmutableArray<NamedTypeSymbol>.Empty;
+        internal override ImmutableArray<Symbol> GetEarlyAttributeDecodingMembers() => GetMembers();
+
+        internal override ImmutableArray<Symbol> GetEarlyAttributeDecodingMembers(string name) => GetMembers(name);
+
+        internal override IEnumerable<FieldSymbol> GetFieldsToEmit() => SpecializedCollections.EmptyEnumerable<FieldSymbol>();
+
+        internal override ImmutableArray<NamedTypeSymbol> GetInterfacesToEmit() => ImmutableArray<NamedTypeSymbol>.Empty;
+
+        internal override IEnumerable<SecurityAttribute> GetSecurityInformation() => null;
+
+        internal override ImmutableArray<NamedTypeSymbol> InterfacesNoUseSiteDiagnostics(ConsList<Symbol> basesBeingResolved = null) => ImmutableArray<NamedTypeSymbol>.Empty;
+
+        internal override void AddSynthesizedAttributes(PEModuleBuilder moduleBuilder, ref ArrayBuilder<SynthesizedAttributeData> attributes)
+        {
+            base.AddSynthesizedAttributes(moduleBuilder, ref attributes);
+
+            AddSynthesizedAttribute(
+                ref attributes,
+                moduleBuilder.Compilation.TrySynthesizeAttribute(WellKnownMember.System_Runtime_CompilerServices_CompilerGeneratedAttribute__ctor));
+
+            AddSynthesizedAttribute(
+                ref attributes,
+                moduleBuilder.SynthesizeEmbeddedAttribute());
+        }
+
+        internal sealed class SourceEmbeddedAttributeConstructorSymbol : SynthesizedInstanceConstructor
```

Comment:
>SynthesizedInstanceConstructor [](start = 73, length = 30)

This type assumes System.Void is available, that should be verified and appropriate diagnostics should be reported.